lib/converter.js

```diff
@@ -64,7 +64,7 @@ async function convert (processor, inputFile, options, timings) {
   }
   const browser = await puppeteer.launch(puppeteerConfig)
   try {
-    const page = await browser.newPage()
+    const page = (await browser.pages())[0]
```

Comment:
If there's no page we should probably call `newPage()`:

```suggestion
    const pages = await browser.pages()
    let page;
    if (pages && pages.length > 0) {
      page = pages[0]
    } else {
      page = await browser.newPage()
    }
```

Maybe it's too defensive... but since it's an array of pages I think we should be careful.
What do you think?

Reply:
Well... there seems to exist a page as soon you open the browser... So I would rely on that? 🤔 . It is a headless browser, so I would not expect that behaviour to change between runs...

Reply:
That's true but the API does not explicitly say that the array will never be empty: https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#browserpages

Who knows maybe for performance or security reasons the "default" page will be removed and `pages()` will return an empty array.
I think it's reasonable to check for the size of the array.

Reply:
Well I guess one would quickly notice during an upgrade :P But fine with me.

Reply:
You're probably right, let's ask a third party :grinning:  
@oncletom @ahus1 do you have an opinion? should we just use `(await browser.pages())[0]`?

Reply:
I don't know, I would favour code clarity over anything.
So, what makes it more readable to you.

I'm not sure to understand what is the benefit of changing the URL instead of opening a new page (clearer code). Does it have an impact on performances or something?

Reply:
It has an impact with the future debug/preview mode where the browser will open the HTML file (non-headless).
In this case, the browser has two tabs which is a bit confusing:

- a default page
- a new page with the HTML file

Performance wise I don't think it matters.